kolibri/core/templatetags/kolibri_tags.py

```diff
@@ -36,15 +39,51 @@ def kolibri_main_navigation():
 
 def render_as_tags(bundle):
     tags = []
-    static = getattr(settings, 'STATIC_URL')
     for chunk in bundle:
-        url = chunk.get('publicPath') or chunk['url']
         if chunk['name'].endswith('.js'):
-            tags.append('<script type="text/javascript" src="{static}/{url}"></script>'.format(static=static, url=url))
+            tags.append('<script type="text/javascript" src="{url}"></script>'.format(url=render_as_url(chunk)))
         elif chunk['name'].endswith('.css'):
-            tags.append('<link type="text/css" href="{static}/{url}" rel="stylesheet"/>'.format(static=static, url=url))
+            tags.append('<link type="text/css" href="{url}" rel="stylesheet"/>'.format(url=render_as_url(chunk)))
     return mark_safe('\n'.join(tags))
 
+
+def render_as_url(chunk):
+    static = getattr(settings, 'STATIC_URL')
+    url = chunk.get('publicPath') or chunk['url']
+    return "{static}{url}".format(static=static, url=url)
+
+
+def render_as_async(bundle):
+    chunks = get_webpack_bundle(bundle, None)
+    async_events = get_async_events(bundle)
+    urls = [render_as_url(chunk) for chunk in chunks]
+    js = 'Kolibri.register_plugin_async("{bundle}", ["{urls}"], {events});'.format(
+        bundle=bundle,
+        urls='","'.join(urls),
+        events=json.dumps(async_events)
+    )
+    return mark_safe('<script>{js}</script>'.format(js=js))
+
+
 @register.simple_tag()
 def frontend_assets(bundle_path, extension=None):
     return render_as_tags(get_webpack_bundle(bundle_path, extension))
+
+
+def frontend_sync(hook):
+    tags = [render_as_tags(get_webpack_bundle(callback(), None)) for callback in hooks.get_callables(hook)]
+    return mark_safe('\n'.join(tags))
+
+
+@register.simple_tag()
+def base_frontend_sync():
+    return frontend_sync(BASE_FRONTEND_SYNC)
+
+
+def frontend_async(hook):
```

Comment:
Yeah - this is where each hook has an 'implicit' API - based on function that its callbacks eventually get passed to.

I agree this could be improved - but it was like that when I got here.